scripts/speakers.js

```diff
@@ -0,0 +1,372 @@
+								'</div></div></div>'));
+
+		$(this).on('change',function (e) {
+
+			// Get file
+			var file = e.target.files[0];
+			if(file.type.search("image/") === 0) {
+				$("#"+parentId+" img").attr('src',''+URL.createObjectURL(file));
+				speakerEditData[""+parentId].fileLoc = file;
+				console.log(speakerEditData);
+			}
+			else {
+				alert("Error : File not an image");
+			}
+		});
+	});
+	inputFile.click();
+}
+var addLogofile = null;
+function getSpeakerLogo() {
+
+	var addLogoInput = $('<input type="file">');
+	addLogoInput.change(function (e) {
+
+		addLogofile =  e.target.files[0];
+		if (addLogofile.type.search("image/") === 0) {
+
+			$("#add-speaker-preview").attr('src',URL.createObjectURL(addLogofile)).show();
+		} else {
+			alert("Error: File not an image");
+			addLogofile = null;
+		}
+	});
+
+	addLogoInput.click();
+}
+
+function addSpeakerToDatabase() {
+
+	var addSpeakerName;
+	var addSpeakerLink;
+	var addSpeakerLogo;
+  var addSpeakerDesc;
+
+	if ($("#add-speaker-name").val()) {
+	 addSpeakerName = $("#add-speaker-name").val();
+	} else {
+		alert("Error : Enter a valid Speaker name");
+		return;
+	}
+
+  if ($("#add-speaker-desc").val()) {
+   addSpeakerDesc = $("#add-speaker-desc").val();
+  } else {
+    alert("Error : Enter a valid Speaker description");
+    return;
+  }
+
+	if ($("#add-speaker-link").val()) {
+		addSpeakerLink = $("#add-speaker-link").val();
+	} else {
+		alert("Error : Enter a valid Speaker link");
+		return;
+	}
+
+	if (addLogofile == null) {
+		alert("Error : Upload a valid Speaker Logo");
+		return;
+	}
+
+
+	var dbRef = firebase.database().ref().child("speakers");
+	var newSpeakerId = dbRef.push().key;
+
+	var storageRef = firebase.storage().ref('speakers-image/' +
+										addLogofile.name);
+
+	var task = storageRef.put(addLogofile);
+
+	task.on('state_changed',
+
+		function progress(snapshot) {
+			$("#save-speaker-progress-bar").show();
+			var percentage = (snapshot.bytesTransferred/snapshot.totalBytes)*100;
+			$('#uploader').css('width',percentage+'%');
+			$('#uploader').attr('aria-valuenow',percentage+'%');
+			$('#uploader').text(percentage+'%');
+		},
+
+		function error(argument) {
+			// body...
+		},
+
+		function complete(argument) {
+			storageRef.getDownloadURL().then(function (url) {
+				addSpeakerLogo = url;
+				dbRef.child(newSpeakerId).set(
+					{
+
+						onClickUrl : addSpeakerLink,
+						imageUrl : addSpeakerLogo,
+						name : addSpeakerName,
+            desc : addSpeakerDesc
+					},
+					function error(errorarg) {
+						if(errorarg){
+							console.error(errorarg);
+						} else {
+
+								//Resetting Add new Speaker
+								$('#add-speaker-btn').click();
+								$('#save-speaker-progress-bar').hide();
+								$('#add-speaker-preview').hide();
+								$('#add-speaker-name').val("");
+                $('#add-speaker-desc').val("");
+								$('add-speaker-link').val("");
+								addLogofile = null;
```

Comment:
add-speaker-link should have # prepended to it. i.e Line 362 should read $('**#**add-apeaker-link').